### PR TITLE
Added nginx reverse proxy example

### PR DESCRIPTION
**Pull request by fbretmounet**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,9 +58,44 @@ iptables -t nat -A PREROUTING -i eth0 -p tcp -d 172.4.0.1 --dport 1883 -j DNAT -
...
+Please note that you can also have the container running in your local network and will not need to modify IPTables. For example if you local network is on 192.168.1.x and your container's IP address is 192.168.1.10, point your local DNS resolver to that IP address and you do not need to add iptable entries.
+
 # Running the app
 
-You will need to run some form of SSL proxy (NGINX, Traefik, etc.) in front of this service. 
+You will need to run some form of SSL proxy (NGINX, Traefik, etc.) in front of this service.
+The SSL service is required and will need to listen on 443 and reverse proxy back to the docker container service on port 5000.
+You will need to generate a self-signed certificate for it. It can be generated with openssl:
+
+...
```
Comment (kruton) [inline README.md:69]: You can use Markdown for this. Replace ... with \`\`\`sh and at the end replace ... with \`\`\`
```suggestion
```bash
```

File: README.md [hunk 2]
```diff
@@ -58,9 +58,44 @@ iptables -t nat -A PREROUTING -i eth0 -p tcp -d 172.4.0.1 --dport 1883 -j DNAT -
...
+  sudo openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /etc/nginx/cert.key -out /etc/nginx/cert.crt 
+...
```
Comment (kruton) [inline README.md:71]:
````suggestion
```
````

File: README.md [hunk 3]
```diff
@@ -58,9 +58,44 @@ iptables -t nat -A PREROUTING -i eth0 -p tcp -d 172.4.0.1 --dport 1883 -j DNAT -
...
+
+If you are using NGINX, your config will look like this:
+
+
```
Comment (kruton) [inline README.md:75]:
````suggestion
```nginx
````

File: README.md [hunk 4]
```diff
@@ -58,9 +58,44 @@ iptables -t nat -A PREROUTING -i eth0 -p tcp -d 172.4.0.1 --dport 1883 -j DNAT -
...
+    ssl_protocols  TLSv1 TLSv1.1 TLSv1.2;
+    ssl_ciphers HIGH:!aNULL:!eNULL:!EXPORT:!CAMELLIA:!DES:!MD5:!PSK:!RC4;
+    ssl_prefer_server_ciphers on;
+
+        location / {
+         proxy_pass http://[CONTAINER_IP]:5000;
+         proxy_redirect http://[CONTAINER_IP]:5000 https://[SERVER_NAME]:443;
+        
+        }
+
```
Comment (kruton) [inline README.md:94]:
````suggestion
```
````

File: README.md [hunk 5]
```diff
@@ -58,9 +58,44 @@ iptables -t nat -A PREROUTING -i eth0 -p tcp -d 172.4.0.1 --dport 1883 -j DNAT -
...
+    ssl_certificate           /etc/nginx/cert.crt;
+    ssl_certificate_key       /etc/nginx/cert.key;
+
+    ssl on;
+    ssl_session_cache  builtin:1000  shared:SSL:10m;
+    ssl_protocols  TLSv1 TLSv1.1 TLSv1.2;
+    ssl_ciphers HIGH:!aNULL:!eNULL:!EXPORT:!CAMELLIA:!DES:!MD5:!PSK:!RC4;
+    ssl_prefer_server_ciphers on;
+
+        location / {
```
Comment (kruton) [inline README.md:89]: Indentation is too great here.
```suggestion
    location / {
```

File: README.md [hunk 6]
```diff
@@ -58,9 +58,44 @@ iptables -t nat -A PREROUTING -i eth0 -p tcp -d 172.4.0.1 --dport 1883 -j DNAT -
...
+         proxy_pass http://[CONTAINER_IP]:5000;
+         proxy_redirect http://[CONTAINER_IP]:5000 https://[SERVER_NAME]:443;
+
```
Comment (kruton) [inline README.md:92]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -58,9 +58,44 @@ iptables -t nat -A PREROUTING -i eth0 -p tcp -d 172.4.0.1 --dport 1883 -j DNAT -
...
+    ssl_session_cache  builtin:1000  shared:SSL:10m;
+    ssl_protocols  TLSv1 TLSv1.1 TLSv1.2;
+    ssl_ciphers HIGH:!aNULL:!eNULL:!EXPORT:!CAMELLIA:!DES:!MD5:!PSK:!RC4;
+    ssl_prefer_server_ciphers on;
+
+        location / {
+         proxy_pass http://[CONTAINER_IP]:5000;
+         proxy_redirect http://[CONTAINER_IP]:5000 https://[SERVER_NAME]:443;
+        
+        }
```
Comment (kruton) [inline README.md:93]: Indentation too great here.

```suggestion
    }
```

File: README.md [hunk 8]
```diff
@@ -58,9 +58,44 @@ iptables -t nat -A PREROUTING -i eth0 -p tcp -d 172.4.0.1 --dport 1883 -j DNAT -
 iptables -t nat -A PREROUTING -i eth0 -p tcp -d 172.4.0.1 --dport 8883 -j DNAT --to 192.168.1.10:8883
 ```
 
+Please note that you can also have the container running in your local network and will not need to modify IPTables. For example if you local network is on 192.168.1.x and your container's IP address is 192.168.1.10, point your local DNS resolver to that IP address and you do not need to add iptable entries.
+
 # Running the app
 
-You will need to run some form of SSL proxy (NGINX, Traefik, etc.) in front of this service. 
+You will need to run some form of SSL proxy (NGINX, Traefik, etc.) in front of this service.
+The SSL service is required and will need to listen on 443 and reverse proxy back to the docker container service on port 5000.
```
Comment (kruton) [inline README.md:66]: Above this we mention the IPTables issue, but then this says you must listen on 443. Perhaps this should be broken out into a `docs/` subdirectory where we discuss individual scenarios.

**Review by kruton — CHANGES_REQUESTED** [review]

**Comment by mbillow** [comment]
I am fine with this being in the README for now but we really should start breaking this content up into the GitHub Wiki. I just went and enabled it for the project and will start setting it up.

**Comment by fbretmounet** [comment]
No longer needed based on adding native support for https.

**Pull request closed without merging**